### PR TITLE
Added JWT support

### PR DESCRIPTION
**Pull request by Lessorg**

### CODE REVIEW

File: src/main/java/test/project/bookshop/controller/AuthenticationController.java [hunk 1]
```diff
@@ -19,12 +24,24 @@
...
     public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto requestDto)
             throws RegistrationException {
         return userService.register(requestDto);
     }
+
+    @Operation(
+            summary = "Login a user",
+            description = "Authenticates a user and returns a JWT token"
+    )
+    @ResponseStatus(HttpStatus.OK)
```
Comment (liliia-ponomarenko) [inline src/main/java/test/project/bookshop/controller/AuthenticationController.java]: it's default response status

File: src/main/java/test/project/bookshop/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    userDetails, null, userDetails.getAuthorities()
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
```
Comment (liliia-ponomarenko) [inline src/main/java/test/project/bookshop/security/JwtAuthenticationFilter.java]: HttpHeader.AUTHORIZATION

File: src/main/java/test/project/bookshop/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        if (bearerToken != null && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
Comment (liliia-ponomarenko) [inline src/main/java/test/project/bookshop/security/JwtAuthenticationFilter.java]: Let’s make it literal like a constant field and after you can use the `length()` method instead 7

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -6,3 +6,5 @@ spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
 spring.jpa.open-in-view=false
+jwt.expiration=400000
+jwt.secret=The_secret_key_must_be_at_least_256_bits_(32_characters)_long
```
Comment (liliia-ponomarenko) [inline src/main/resources/application.properties:10]: better generate random secret

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
Good job! Let’s improve your solution ;)

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**